api/java/org/openyolo/api/internal/ActivityResult.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 The OpenYOLO Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openyolo.api.internal;
+
+import android.content.Intent;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.Nullable;
+
+/**
+ * Parcelable representation of an activity result.
+ */
+public class ActivityResult implements Parcelable {
+
+    /**
+     * Parcelable creator for {@link ActivityResult}.
+     */
+    public static final Parcelable.Creator<ActivityResult> CREATOR = new Creator();
+
+    private int mResultCode;
+    private Intent mData;
+
+    /**
+     * Returns a new instance of an {@link ActivityResult} from the given result code and intent
+     * data.
+     */
+    public static ActivityResult of(int resultCode, @Nullable Intent data) {
+        return new ActivityResult(resultCode, data);
+    }
+
+    private ActivityResult(int resultCode, @Nullable Intent data) {

```

Comment:
I don't believe onActivityForResult will ever pass a null intent data object, lets not allow it here either.